plutus-core/plutus-core/src/PlutusCore/Builtin/Runtime.hs

```diff
@@ -34,6 +35,13 @@ data RuntimeScheme val (args :: [GHC.Type]) res where
         :: RuntimeScheme val args res
         -> RuntimeScheme val args res
 
+-- we use strictdata, so this is just for the purpose of completeness
+instance NFData (RuntimeScheme val args res) where
+    rnf r = case r of
+        RuntimeSchemeResult    -> rwhnf r
```

Comment:
I think this is redundant since you already cased on `r`, but whatever

Reply:
i know, but i followed the practice of NFData where they also define
```
instance NFData () where
    rnf = rwhnf
```

Reply:
In the `()` instance `rwhnf` is not redundant: it catches the `undefined :: ()` case. Since you match on `r` explicitly, that also catches the `undefined` case, so there's no need to call `rwhnf` over `r` afterwards.